subwabbit/blocking.py

```diff
@@ -286,3 +288,31 @@ def _get_predictions_from_vowpal(self, detailed_metrics=None, debug_info=None):
         if detailed_metrics:
             detailed_metrics['receiving_lines_time'].append((time.perf_counter(), time.perf_counter() - t0))
         return predictions
+
+
+class VowpalWabbitPLTProcess(VowpalWabbitProcess):
+    """
+    Class representing Vowpal Wabbit model. It runs ``vw`` command
+    through subprocess library and communicates through pipes.
+
+    The only difference from VowpalWabbitProcess
+    is the support of Probablistic Label Tree functionality in prediction.
+    https://vowpalwabbit.org/blog/vowpalwabbit-8.9.0.html#probabilistic-label-tree
+
+    Returned predictions are strings rather than floats.
+    """
+
+    def _get_predictions_from_vowpal(self, detailed_metrics=None, debug_info=None):  # pylint:  disable=unused-argument
+        # There should be check whether instance is not in write-only mode,
+        # but for predictions we have to be damn fast so eventually
+        # let this method fail on calling pop(0) on empty list
+        num_predictions = self.unprocessed_batch_sizes.pop(0)
+        t0 = time.perf_counter()
+        predictions = []
+        received_line = None
+        for _ in range(num_predictions):
+            received_line = self.vw_process.stdout.readline().strip()
+            predictions.append(received_line.decode("utf-8"))
```

Comment:
As it now returns something else then float, type hints for `predict` method are violated, can you please change the return type hint to `Iterable[Union[float,str]]` [here](https://github.com/jakac/subwabbit/pull/3/files#diff-45689bc5ffa1c7f94e1f4314c7bf6d44ba1dbd954d55e18d167a736135751719R102) and [here](https://github.com/jakac/subwabbit/blob/master/subwabbit/base.py#L258)?

Reply:
Thanks, fixed!